src/Acornima.Extras/ScopeInfo.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Acornima.Ast;
+
+namespace Acornima;
+
+public sealed class ScopeInfo
+{
+    private readonly int _id;
+    private readonly int _currentVarScopeId;
+    private readonly int _currentThisScopeId;
+
+    public ScopeInfo(Node node, in Scope scope, ReadOnlySpan<Scope> scopeStack)
+    {
+        AssociatedNode = node;
+        _id = scope.Id;
+        _currentVarScopeId = scope.CurrentVarScopeIndex == scopeStack.Length ? scope.Id : scopeStack[scope.CurrentVarScopeIndex].Id;
+        _currentThisScopeId = scope.CurrentThisScopeIndex == scopeStack.Length ? scope.Id : scopeStack[scope.CurrentThisScopeIndex].Id;
+        VarNames = scope.VarNames.ToArray();
```

Comment:
I noticed that `scope.VarNames` can have the same variable name repeated multiple times, this makes sense after checking the parser, however, I think this will affect the performance of consumers that loop through variable names. 

Should we Keep it as-is or apply a Distinct? 

Context: the parser carries var definitions bottom to top, parent nodes will have on their var names, the variables defined (or re-defined) by the child nodes.

Example:
```js
var x = "a";  
switch(x) {
  case "a":
  case 2:
  let y = 3;
  var x = 1;  
  case 3:
    console.log(x); 
    break;
}
```

Is parsed as:
```
Program (x,x)
 └─SwitchStatement (x,y)
```

Reply:
I did some benchmarks to see if we can remove duplicate names during parsing without a performance penalty. I found that we can't; the current implementation, which doesn't care about duplicates names, has the best performance characteristics on average.

Of course, this doesn't prevent us from getting rid of the the duplicate names when opting in to scope info tracking (i.e. when creating `ScopeInfo` objects, which must create a copy of the variables name lists anyway).

I added another commit that [implements this](https://github.com/adams85/acornima/pull/13/files#diff-0b03b54549bcc6d5d0fbc9f5dea9dc12aad99969544417fa6aa37e8037e13c5fR25-R29).